h2o-py/tests/testdir_algos/gam/pyunit_PUBDEV_7367_random_gridsearch_specific.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import division
+from __future__ import print_function
+from past.utils import old_div
+import sys
+sys.path.insert(1, "../../../")
+import h2o
+from tests import pyunit_utils
+from h2o.estimators.gam import H2OGeneralizedAdditiveEstimator
+from h2o.grid.grid_search import H2OGridSearch
+
+# In this test, we check to make sure that a random discrete grid search on a GAM functions correctly.
+# The test searches over 5 parameters, alpha, lambda, scale, gam_columns, and num_knots.
+# The test then compares the results of the grid search models with the models we created
+# by manually searching over the hyperspace.
+# If the coefficients do not match or an incorrect number of models is generated, the test throws an assertion error.
+class test_random_gam_gridsearch_specific:
+
+    h2o_data = []
+    myX = []
+    myY = []
+    h2o_model = []
+    search_criteria = {'strategy': 'RandomDiscrete', "max_models": 8, "seed": 1}
+    hyper_parameters = {'scale': [[1, 1], [2, 2], [0.01], [0.05]], 'gam_columns': [["C6", "C7"], ["C6"]], 'num_knots': [[5, 5], [6, 6], [5], [6]]}
+    manual_gam_models = []
```

Comment:
Karthik:  Can you add a gam-column set of three columns?  and add two choices for it in scale and num_knots?  Just want to check when things get bigger, it still works.